lib/protocol/Protocol.js

```diff
@@ -336,6 +336,14 @@ Protocol.prototype._dequeue = function(sequence) {
     return;
   }
 
+  if (sequence !== this._queue[0]) {
+    var idx = this._queue.indexOf(sequence);
+    if (idx > 0) {
```

Comment:
>= ?

Reply:
`idx` will never be `0` as `sequence !== this._queue[0]`. The sequence in `this._queue[0]` is the one that is currently being executed, isn't it?

Or you mean to change it to `idx >= 1`?

Reply:
My bad, missed previous line.

Btw maybe use separate method for removing sequence from queue? Its seems current method executes next sequence, but im not sure.

Reply:
@maximelkin I don't call `Protocol._dequeue(sequence)` directly. It is called from handler of `Sequence.emit('end')` emited from `Query.end()` called in my `Query.dequeue()`. I wanted to use the native query ending mechanism, so the query will call its `_callback` (with nothing) after being cancelled.

But yes, if we think it would be better, we could short circuit the native query ending mechanism, and just remove the query from the queue without calling `Query.end()` and emitting `Sequence.emit('end')`. But then the query `_callback` would never be called. I'm not sure if it would be good or not.